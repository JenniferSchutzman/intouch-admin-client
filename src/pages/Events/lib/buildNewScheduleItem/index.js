import { merge } from "ramda"
import moment from "moment-timezone"
import shortId from "shortid"


/**
 * buildNewScheduleItem - creates new schedule 
 * @param {Object} selectedTime - time selected
 * @param {String} eventDate - date of event
 * @param {Object} formData - data from form
 */
const buildNewScheduleItem = (selectedTime, eventDate, formData) => {
  const { hour, minute, timeOfDay } = selectedTime
  const timeString = `${hour}:${minute} ${timeOfDay}`

  const time = {
    string: timeString,
    unix: moment(`${eventDate} ${timeString}`, "YYYY-MM-DD h:mm a").unix()
  }

  const eventData = { time, id: shortId() }
  return merge(formData, eventData)
}

export default buildNewScheduleItem
