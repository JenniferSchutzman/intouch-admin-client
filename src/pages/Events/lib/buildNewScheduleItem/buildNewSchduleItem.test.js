import buildNewScheduleItem from "./"
import { omit } from "ramda"

test("should build new schedule item", () => {
  const result = buildNewScheduleItem(selectedTime, eventDate, formData)
  expect(omit(["id"], result)).toMatchSnapshot()
})

var selectedTime = {
  open: true,
  hour: "12",
  minute: "00",
  timeOfDay: "pm",
  endTime: {
    hour: "6",
    minute: "00",
    timeOfDay: "pm",
    hasEndTime: true
  }
}
var eventDate = "2017-11-15"

var formData = {
  name: "test name"
}
