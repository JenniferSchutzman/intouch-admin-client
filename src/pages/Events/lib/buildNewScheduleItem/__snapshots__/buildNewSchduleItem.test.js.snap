// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build new schedule item 1`] = `
Object {
  "name": "test name",
  "time": Object {
    "string": "4:00 pm",
    "unix": 1510779600,
  },
}
`;
